### ISSUE TITLE
RenderableSatellites lines/points appearance

### ISSUE THREAD

**Issue by hedbergj** [OpenSpace/OpenSpace#1472]
In the Renderable->Appearance tab for a group of satellites, the standard trail options like line/points/line+points are there, but only the line option seems to be functional. 

What might be more useful, rather than the normal trail points option, is a line or dots selector. So that displays like this:

http://astria.tacc.utexas.edu/AstriaGraph/

can be done simply. 

Basically, the option to just show the satellites from TLE files like simple stars (billboard texture or even just a little not-too-smart circle) would be good to have. 

(Linewidths >1 on Mac don't function so sometimes it's hard to get any good quality images that include trails.)

**Comment by alexanderbock** [comment]
Agreed!   Rendering them as point sources is not super trivial, but I think it can be done.  What do you thing @GPayne ?

**Comment by hedbergj** [comment]
(hopefully not a true 'point source' then it might be hard to see :) 

I was thinking that whatever was used in the normal RenderableTrails to make the points style display could just be recycled into renderableorbitalkepler, but instead of showing multiple points spaced along the trail, just one at the start.

**Comment by micahnyc** [comment]
Along with point sources, we want more control over the 'linefade' the current values do not let the user go 'low' enough to make a short trail. On the RenderableSattelites you can, but the RenderableSmallBodies seem to have too large of a trail to make it fade down to a small point. 

For things like comets, I think it would still be valuable to have the lines+points options if we can use it to show t he relative velocity of a comet for the portion of it's trail as it goes around the sun.

**Comment by GPayne** [comment]
I agree that the rendering mode menu presents line/point/lines+points options, and this should be addressed since only lines currently works.
With satellites or asteroids, you can get down to almost a point with a stub trail using appearance->line fade. I like the little directional hint that it provides. The only time this seems to be a problem is asteroids with huge orbits.
I'll look into the ease of adding points, and base the priority on that. We used to do points with a small billboard, but I think a GL point would be better. It would be nice to have just points when you are rendering a high number of objects such as in the AstriaGraph link.

**Comment by hedbergj** [comment]
I guess I was hoping to be able to show a view like this (in addition to the many object astriagraph view): 

https://www.youtube.com/watch?v=MBssP7awKRM

Even with line fade = 0, they are still pretty long (and believe me, I tried every workaround I could think of)  

![OpenSpace_000001](https://user-images.githubusercontent.com/7926907/106190379-942e5900-6177-11eb-9e3a-a26e6cdefe37.png)

(And certainly the trails are good for somethings, I just also like the idea of trying to show them more or less as they would appear, in keeping with the larger goals of OS. )

**Comment by alexanderbock** [comment]
Sorry for not being clear;  I definitely consider this to be a bug to be fixed. . the satellites/asteroids should be renderable as points.

In the meantime, if you don't shy away from it:
In the `module/space/shaders/debrisVis_fs.glsl` file:
In line 55 I added:
`    if (vertexDistance < 0.003) { discard;}`
so that it looks like:
```
    float vertexDistance = periodFraction_f - offsetPeriods;
    if (vertexDistance < 0.003) { discard;}
    float vertexDistance_f = float(vertexDistance);
```

and then further down in line 84:
`    frag.color.a = 1.0;`
so that it looks like:
```    frag.color = vec4(color, fade * opacity);
    frag.color.a = 1.0;
    frag.depth = vs_position_w;
```

Then all of the trails that have a line fade of 0 will be roughly point-like.  You need to restart OpenSpace after editing the file the first time or you will get error messages, but for tweaking the 0.003 value you don't need to restart for every change.

![Uploading OpenSpace_000000.png…]()

**Comment by GPayne** [comment]
Nice. I tried to hack something like this yesterday but I couldn't get it to look as good. Increasing the line width a little can help visibility, but unfortunately starts to distort if you zoom out a lot.

**Comment by micahnyc** [comment]
Related issue: https://github.com/OpenSpace/OpenSpace/issues/1212

**Comment by hedbergj** [comment]
Nice example here: https://www.nytimes.com/interactive/2023/07/28/business/starlink.html
<img width="1177" alt="trails-as-points" src="https://github.com/OpenSpace/OpenSpace/assets/7926907/03041fcf-8abe-4079-ada7-abf156cb1236">

**Comment by alexanderbock** [comment]
LineWidth and PointWidth properties are linked for trails and orbits when they shouldn't be.   For RenderableSatellites the render option doesn't work at all

**Comment by alexanderbock** [comment]
@hedbergj 's example in the PR

![image](https://github.com/OpenSpace/OpenSpace/assets/4921724/00ff2411-99b2-4cfc-9acf-530dd06b1cdc)

**Comment by alexanderbock** [comment]
and from the surface:
![image](https://github.com/OpenSpace/OpenSpace/assets/4921724/a8da415c-687e-4223-b32d-99e7031dff35)

**Comment by Arohdin** [comment]
Resolved in #3193
